### PR TITLE
Revert "ENG-17626 Use C++11 TLS types to replace pthread (#6495)"

### PR DESCRIPTION
**Pull request by n3world**
This reverts commit 0ae47f0cbdd983656582c9d243ce17222702d836, which was
causeing TestImportSuite to wedge the JVM during shutdown.

### CODE REVIEW

**Comment by kaikaizi** [comment]
The TestImportSuite test passes on my machine and doesn't seem to have failed in http://ci.voltdb.lan/view/Branch-jobs/view/branch-master/job/branch-2-community-junit-master/.
How did you reproduce that test error?

File: src/ee/common/SynchronizedThreadLock.cpp [hunk 1]
```diff
@@ -184,7 +188,12 @@ bool SynchronizedThreadLock::countDownGlobalTxnStartCount(bool lowestSite) {
         {
             std::unique_lock<std::mutex> g(s_sharedEngineMutex);
             if (--s_globalTxnStartCountdownLatch != 0) {
-                s_wakeLowestEngineCondition.wait(g);
+                // NOTE: using std::condition_variable::wait()
+                // methods would hang TestImportSuite. So we have
+                // to use pthread_ methods. Using
+                // std::condition_variable only for its RAII.
```
Comment (n3world) [inline src/ee/common/SynchronizedThreadLock.cpp]: Comment is out of date

File: src/ee/common/SynchronizedThreadLock.cpp [hunk 2]
```diff
@@ -33,8 +33,8 @@ namespace voltdb {
 
 // Initialized when executor context is created.
 std::mutex SynchronizedThreadLock::s_sharedEngineMutex{};
-std::condition_variable SynchronizedThreadLock::s_sharedEngineCondition;
-std::condition_variable SynchronizedThreadLock::s_wakeLowestEngineCondition;
+pthread_cond_t SynchronizedThreadLock::s_sharedEngineCondition;
+pthread_cond_t SynchronizedThreadLock::s_wakeLowestEngineCondition;
```
Comment (n3world) [inline src/ee/common/SynchronizedThreadLock.cpp]: You could initialize the variables here using PTHREAD_COND_INITIALIZER and remove the init in the create methoc

File: src/ee/common/SynchronizedThreadLock.h [hunk 3]
```diff
@@ -64,8 +64,8 @@ class SynchronizedThreadLock {
 #endif
     static bool s_holdingReplicatedTableLock;
     static std::mutex s_sharedEngineMutex;
-    static std::condition_variable s_sharedEngineCondition;
-    static std::condition_variable s_wakeLowestEngineCondition;
+    static pthread_cond_t s_sharedEngineCondition;
```
Comment (n3world) [inline src/ee/common/SynchronizedThreadLock.h]: Import of `condition_variable` was not removed.

**Pull request merged**